pkg/server/server.go

```diff
@@ -459,8 +463,10 @@ func (s *ProxyServer) serveRecvFrontend(stream client.ProxyService_ProxyServer,
 			if err := backend.Send(pkt); err != nil {
 				// TODO: retry with other backends connecting to this agent.
 				klog.ErrorS(err, "CLOSE_REQ to Backend failed", "serverID", s.serverID, "connectionID", connID)
+			} else {
+				klog.V(5).Infoln("CLOSE_REQ sent to backend", "serverID", s.serverID, "connectionID", connID)
+				return
```

Comment:
@mainred I think it might be simpler to scope this PR to only include DialResponse containing random IDs and exclude the fix for sending CLOSE_REQ twice in a separate PR.  @mihivagyok also has a PR out regarding this https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/322